crates/holochain_p2p/tests/tests/node_messaging.rs

```diff
@@ -263,6 +266,119 @@ async fn test_remote_signal() {
     .unwrap();
 }
 
+fn test_dht_op(authored_timestamp: holochain_types::prelude::Timestamp) -> DhtOpHashed {
+    let mut create = ::fixt::fixt!(Create);
+    create.timestamp = authored_timestamp;
+
+    let op = DhtOp::from(ChainOp::StoreRecord(
+        ::fixt::fixt!(Signature),
+        Action::Create(create),
+        RecordEntry::Present(::fixt::fixt!(Entry)),
+    ));
+    DhtOpHashed::from_content_sync(op)
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_publish() {
+    let dna_hash = DnaHash::from_raw_36(vec![0; 36]);
+    let space = dna_hash.to_k2_space();
+    let handler = Arc::new(Handler::default());
+
+    let (_agent1, hc1) = spawn_test(dna_hash.clone(), handler.clone()).await;
+    let (_agent2, hc2) = spawn_test(dna_hash.clone(), handler.clone()).await;
+
+    hc1.test_set_full_arcs(space.clone()).await;
+    hc2.test_set_full_arcs(space.clone()).await;
+
+    tokio::time::timeout(std::time::Duration::from_secs(20), async {
+        loop {
+            tokio::time::sleep(std::time::Duration::from_millis(1)).await;
+
+            let op = test_dht_op(holochain_types::prelude::Timestamp::now());
+            let op_hash = op.as_hash().clone();
+
+            hc2.test_kitsune()
+                .space(dna_hash.to_k2_space())
+                .await
+                .unwrap()
+                .op_store()
+                .process_incoming_ops(vec![bytes::Bytes::from(
```

Comment:
Would it be better for tests that we keep a direct handle to the store around, so that we can write to it without calling this method?

Reply:
Is there any reason it's bad to call this function?

Reply:
It has side effects and is meant to process ops that the node received through gossip or publish. This test (and likely others) wants to write ops that the node created to the store and publish them.

Reply:
Is it okay to merge as-is and address this the next time a test like this is written?

Reply:
Yes, sure. Please a todo here.

Reply:
Done: https://github.com/holochain/holochain/pull/4788/commits/93364cfc298532ce1b5992c42fd26c7bbe7b6e99